examples/step-64/step-64.cu

```diff
@@ -304,6 +307,15 @@ namespace Step64
   }
 
 
+
+  template <int dim, int fe_degree>
+  void HelmholtzOperator<dim, fe_degree>::initialize_dof_vector(
+    LinearAlgebra::distributed::Vector<double, MemorySpace::CUDA> &vec) const
+  {
+    mf_data.initialize_dof_vector(vec);
```

Comment:
As far as I see it, this function does not zero out the entries. Inside it calls `reinit(const std::shared_ptr<const Utilities::MPI::Partitioner> &partitioner)`, which does not  have the flag `const bool omit_zeroing_entries = false`. I do not know if this helps. Everything else looks good (from a standpoint that I do not have access to a compiler).

Reply:
This might be a problem. All the constrained entries need to be zero out.

Reply:
The [function `reinit(Partitioner)`](https://github.com/dealii/dealii/blob/master/include/deal.II/lac/la_parallel_vector.templates.h#L511-L536) should indeed zero the vector, including the ghosts, as it calls `operator=(Number())`. I also checked that the `zero_out_ghosts()` function is doing the right thing.